### PR TITLE
fix race condition in ReplicatorDeltaSpec (#22737)

### PR DESCRIPTION
**Pull request by gosubpl**
Refs #22737

### CODE REVIEW

**Comment by gosubpl** [comment]
This fixes a very rare issue when one of the replicas from set `(second, third, fourth)` is delayed so much, that it gets read propagated from `first`.

**Comment by raboof** [comment]
LGTM

File: akka-distributed-data/src/multi-jvm/scala/akka/cluster/ddata/ReplicatorDeltaSpec.scala [hunk 1]
```diff
@@ -241,6 +241,7 @@ class ReplicatorDeltaSpec extends MultiNodeSpec(ReplicatorDeltaSpec) with STMult
       val p = TestProbe()
       deltaReplicator.tell(Get(KeyD, ReadLocal), p.ref)
       p.expectMsgType[GetSuccess[ORSet[String]]].dataValue.elements should ===(Set("a", "A"))
+      enterBarrier("read-1")
```
Comment (patriknw) [inline akka-distributed-data/src/multi-jvm/scala/akka/cluster/ddata/ReplicatorDeltaSpec.scala]: ah, some nodes get the update from below

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@raboof please use the Review Changes button for the LGTM, because then it shows up with a nice check box and such

**Comment by patriknw** [comment]
Thanks @gosubpl

**Pull request merged**